### DISCUSSION TITLE
Cross-compiling for the Nintendo Wii

### DISCUSSION THREAD

**Discussion by mardy** [Neverball/neverball#373] [General]
Hi all!
  I'm trying to port Neverball to the Nintendo Wii. I've got into this problem, that happens both with the current git version and with 1.6.0:
```
./mapc data/map-misc/checkers.map data
./mapc data/map-misc/timer-test_2.map data
./mapc: 1: Syntax error: "(" unexpected
make: *** [Makefile:410: data/map-misc/checkers.sol] Error 2
make: *** Waiting for unfinished jobs....
```
After some initial puzzling, I realized that mapc too is being built for the Nintendo Wii:
```
$ file mapc 
mapc: ELF 32-bit MSB executable, PowerPC or cisco 4500, version 1 (SYSV), statically linked, with debug_info, not stripped
```
which is probably wrong, if this program is meant to be run during the build process, on the host. Do you have any suggestions on how to fix this?

**Comment by parasti** [comment]
mapc is the Neverball map compiler. The generated SOL files are architecture-independent. I suggest you build them separately with `make sols` on whatever system is convenient for you, find and save the `*.sol` files, and then cross-compile the rest of the game without the SOLs with `make -o sols`, then copy the saved SOLs into your cross-compiled tree.
